pkg/providers/instance/instance.go

```diff
@@ -435,6 +462,14 @@ func (p *Provider) launchInstance(
 		return nil, nil, azErr
 	}
 
+	if p.provisionMode == options.ProvisionModeBootstrappingClient {
+		err = p.createCSExtension(ctx, resourceName, launchTemplate.AgentBakerCSE, launchTemplate.IsWindows)
+		if err != nil {
+			// This should fall back to cleanupAzureResources
```

Comment:
Should we be logging the CSE Failure at least? Or creating k8s events for CSE Failure?

Reply:
Logged from within createCSExtension